### ISSUE TITLE
Mitie QA: `MitieIntentClassifier` does not classify the intent correctly

### ISSUE THREAD

**Issue by ancalita** [RasaHQ/rasa#10109]
**Steps to replicate**:
- `rasa init` to a new project
- amend config file to this content:
```
recipe: default.v1

# Configuration for Rasa NLU.
# https://rasa.com/docs/rasa/nlu/components/
language: en

pipeline:
- name: "MitieNLP"
  model: "data/total_word_feature_extractor.dat"
- name: "MitieTokenizer"
- name: "MitieFeaturizer"
- name: "MitieEntityExtractor"
- name: "MitieIntentClassifier"

# Configuration for Rasa Core.
# https://rasa.com/docs/rasa/core/policies/
policies:
# # No configuration for policies was provided. The following default policies were used to train your model.
# # If you'd like to customize them, uncomment and adjust the policies.
# # See https://rasa.com/docs/rasa/policies for more information.
#   - name: MemoizationPolicy
#   - name: RulePolicy
#   - name: UnexpecTEDIntentPolicy
#     max_history: 5
#     epochs: 100
#   - name: TEDPolicy
#     max_history: 5
#     epochs: 100
#     constrain_similarities: true
```
- `rasa train`
- `rasa shell` leads to the following story, where `utter_happy` is replaced by `utter_greet` at the end:
```
Your input -> Hi
Hey! How are you?
Your input -> Feeling sad
Here is something to cheer you up:
Image: https://i.imgur.com/nGF1K8f.jpg
Did that help you?
Your input -> Yes
Hey! How are you?
```
**Debug log**: 
```
2021-11-08 12:56:32 DEBUG rasa.core.processor - Received user message 'Yes' with intent '{'name': 'greet', 'confidence': 0.09738887070403174}' and entities '[]'
```

**Comment by joejuzl** [comment]
Are these steps sufficient to replicate? The titles says it's the `MitieIntentClassifier` which is not included in the config of `rasa init`.

**Comment by joejuzl** [comment]
> @joejuzl my bad, this was part of the Mitie test, so the pipeline should only have Mitie components.

Can you include the config in the description please?

**Comment by usc-m** [comment]
I can't actually replicate this - I've tested this in a fresh project, an old project reconfigured, then also deleted that and re-inited it and couldn't see this behaviour. Going to take another look tomorrow more deeply to see if there's maybe something else going on?

**Comment by joejuzl** [comment]
I would compare the data in a `Message` and see where it differs e.g. tokens, features etc.

**Comment by ka-bu** [comment]
Another datapoint: Doesn't throw any error for me on current main

**Comment by TyDunn** [comment]
@usc-m what is our path forward here?

**Comment by usc-m** [comment]
@ka-bu and I have tried to reproduce this (and asked a few other people to try on their machines) and we can't seem to find anything that's causing it directly. The produced model, if shared with someone, does contain the error but it's not obvious what's actually causing it as we've tried substituting out most of the relevant mitie files, tried clearing the cache, inspecting the cache, re-creating the whole project etc. It's also been tested by @ancalita on 2.8 which does not display the issue (even though only her machine displays the issue on 3.0), so we know it's definitely related to the 3.0 changes but it is not clear where it's being introduced.

Closing this for now as can't reproduce
